sky/packages/sky/lib/src/material/scaffold.dart

```diff
@@ -4,92 +4,87 @@
 
 import 'dart:ui' as ui;
 
+import 'package:flutter/rendering.dart';
 import 'package:flutter/widgets.dart';
 
 import 'constants.dart';
 import 'material.dart';
 
+const int _kToolBarIndex = 1;
+const int _kBodyIndex = 0;
+
+// This layout has the same effect as putting the toolbar and body in a column
+// and making the body flexible. What's different is that in this case the
+// toolbar appears -after- the body in the stacking order, so the toolbar's
+// shadow is drawn on top of the body.
+class _ToolBarAndBodyLayout extends MultiChildLayoutDelegate {
+  void performLayout(Size size, BoxConstraints constraints, int childCount) {
+    assert(childCount == 2);
+    final BoxConstraints toolBarConstraints = constraints.loosen().tightenWidth(size.width);
+    final Size toolBarSize = layoutChild(_kToolBarIndex, toolBarConstraints);
+    final double topPadding = ui.window.padding.top;
+    final double bodyHeight = size.height - toolBarSize.height - topPadding;
+    final BoxConstraints bodyConstraints = toolBarConstraints.tightenHeight(bodyHeight);
+    layoutChild(_kBodyIndex, bodyConstraints);
+    positionChild(_kToolBarIndex, new Point(0.0, topPadding));
```

Comment:
This is tricky because I want the scaffold to respect the toolbar's intrinsic size but, in this case, we need to override that size so that the toolbar extends up and behind the status bar. On Ian's suggestion I've handled that by giving ToolBar a withSizeOffsets(EdgeDims offsets) method that asks the toolbar to expand its size in (potentially) all directions. The Scaffold uses this to grow the toolbar's height.